src/_tkagg.cpp

```diff
@@ -100,39 +90,37 @@ static int PyAggImagePhoto(ClientData clientdata, Tcl_Interp *interp, int
     }
 
     /* check for bbox/blitting */
-    if (sscanf(argv[4], SIZE_T_FORMAT, &bboxl) != 1) {
-        TCL_APPEND_RESULT(interp, "error casting pointer", (char *)NULL);
+    bbox_parse = sscanf(argv[4], BBOX_FORMAT, &x1, &x2, &y1, &y2);
+    if (bbox_parse == 4) {
+        has_bbox = true;
+    }
+    else if ((bbox_parse == 1) && (x1 == 0)){
+        has_bbox = false;
+    } else {
+        TCL_APPEND_RESULT(interp, "illegal bbox", (char *)NULL);
         return TCL_ERROR;
     }
-    bboxo = (PyObject *)bboxl;
-
-    if (bboxo != NULL && bboxo != Py_None) {
-        agg::rect_d rect;
-        if (!convert_rect(bboxo, &rect)) {
-            return TCL_ERROR;
-        }
-
-        has_bbox = true;
 
-        destx = (int)rect.x1;
-        desty = srcheight - (int)rect.y2;
-        destwidth = (int)(rect.x2 - rect.x1);
-        destheight = (int)(rect.y2 - rect.y1);
+    if (has_bbox) {
+        int srcstride = wdata * 4;
+        destx = (int)x1;
+        desty = (int)(hdata - y2);
+        destwidth = (int)(x2 - x1);
+        destheight = (int)(y2 - y1);
         deststride = 4 * destwidth;
 
-        destbuffer = new agg::int8u[deststride * destheight];
+        destbuffer = new uint8_t[deststride * destheight];
```

Comment:
Why change away from the agg types here?

Reply:
Previously there was a conversion from a ``PyArrayObject *`` (in other words ndarray) to a data pointer, width, height, etc via py_converters. Now I pass that information directly from python. I no longer need the "py_converters.cpp" and ""py_converters.h" since there is no use of ndarray in the c++ code. Should I leave the include for this one type definition?

Reply:
I am concerned about the corner case where `unint8_t` and `agg:int8u` are different underlying types.  Not sure how likely that actually is, hence leaning towards being conservative about changing it.

Reply:
I wonder what different types you think these can be.
I would also bet that there must be parts of the code where we assume that this is interconvertible with np.uint8, or that this fits in a single byte... doesn't leave you a lot of options!